crates/communication/src/server/parameters/mod.rs

```diff
@@ -0,0 +1,33 @@
+use serde_json::Value;
+
+use crate::messages::{ParametersRequest, Path};
+
+use super::Client;
+
+pub mod directory;
+pub mod storage;
+pub mod subscriptions;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) struct ClientRequest {
+    pub request: ParametersRequest,
+    pub client: Client,
+}
```

Comment:
Since this type fulfills the same purpose as `Outputs::ClientRequest` and is the same except for one field, perhaps just make it generic over the request type?

Reply:
Done